library(tidyverse)
library(scales)

txt<-"Abstract. Mining rare variant signals from the cancer genome is relevant to a variety of clinical problems like classifying cancers of unknown primary site. So- matic variant mutation analysis has traditionally been restricted to cancer- associated genes with frequent occurrences, neglecting potential signals en- coded in the vast “hidden genome” of rare and hitherto unseen mutations. Indeed, future tumor samples are bound to generate hitherto unseen muta- tions which may contain clinically-relevant signals. Smoothed Good-Turing frequency estimation is an intriguing statistical method that uses mutation richness to estimate probabilities of encountering hitherto unseen mutations. While previous research illustrated the potential of this method, analysis was restricted to the level of frequently mutated genes, omitting from considera- tion any signal in more sparsely mutated genes (which encompass the prepon- derance of the cancer exome). To include a wider cross-section of the cancer exome, this thesis explores the possibility of using de-sparsification strate- gies to aggregate mutation data in thoughtfully-constructed gene groups, for which Good-Turing probabilities can be calculated reliably. This thesis presents two de-sparsification strategies: 1) a higher-variance method that learns gene groupings directly from somatic mutation probability patterns and 2) a higher-bias method that aggregates mutation data within known biological pathways. Generating gene groups that contain cancer type spe- cific hitherto unseen mutation probabilities may improve the ability to har- ness the “hidden genome” of unseen somatic mutation for important clinical tasks. Cancer: A Mutational Malady. Mutation is the unifying hallmark of cancer etiology, though mutation is a remarkably heterogeneous process that varies between cancer types and between patients. Moreover, the vast majority of somatic variant muta- tions (the most abundant mutations in human cancer) are extremely rare in databases of sequenced cancer genomes. This chapter provides an overview of cancer genetics, defines relevant terminology, and lays out the overarching, biomedically-oriented goal of this thesis: extracting clinically relevant signals from the abundance of rare mutation in the cancer genome. Human cancer is a tremendous burden on public health on a global scale. A brief overview of cancer genetics. Even with the advent of modern medicine and improved health care, can- cer has proven an especially difficult malady to treat. Indeed, cancer is the second leading cause of death worldwide (Hassanpour & Mohammadamin 2017). The difficulty of diagnosing and treating cancer lies in its many dif- ferent causes and forms. Cancer is an umbrella term referring to more than 277 different kinds of disease (Hassanpour & Mohammadamin 2017) and is the result of a confluence of environmental and genetic factors (Lodish et al. 2003). However, the underlying cause of cancer is an accumulation of inher- ited or acquired alterations to an individual’s DNA. For the purposes of this thesis, I will focus on the acquired molecular alterations that cause cancer, rather than the epidemiological or behavioral factors that increase disease risk. Cancer is a loss of cellular regulation caused by an accumula- tion of mutations to DNA. Cancer development is based on two processes. First, the continuous, more-or-less random acquisition of mutation in indi- vidual cells throughout their lifetimes. Second, natural selection acting on the phenotype conferred by those mutations (Stratton et al. 2009). In most cases, random mutations are either neutral, meaning they have no impact on an individual’s fitness, or deleterious, which disposes mutated cells to negative selection. However, in some cases, mutation can confer a growth advantage to cells, which allows them to proliferate uncontrollably, spread to other tissues, and cause disease. There are two main classes of mutations in human cancer. Non-heritablemutations that are obtained through a cell’s lifetime are termed somatic mutations. Somatic mutations comprise the majority of cancer causing ge-
netic alterations. Germline, or heritable, mutations are passed down through
family pedigrees (Stratton et al. 2009). There are many different kinds of
somatic alterations, ranging from relatively simple single nucleotide DNA
changes to exotic rearrangements of large DNA regions. The catalogue of so-
matic mutation is a rich resource to understand the etiology and mutational
processes acting in human cancers (Stratton et al. 2009). For the purposes of
this thesis, I will primarily focus on the most common mutation type: single
nucleotide somatic alterations, or somatic variant mutations that occur in the protein-coding, or exonic regions of the genome. Each mutation in cancer may be classified based on its impact vis-a`-vis oncogenesis (cancer development). Mutations can either be classified as drivers or passengers. Drivers, or causal mutations, confer a growth advan- tage that allows cancer cells to proliferate unchecked. Passenger mutations are biologically inert and do not contribute to tumor development (Stratton et al. 2009). It is important to note that a single mutation is rarely ever sufficient to induce oncogenesis alone. Most scientists agree on a “multi- hit” hypothesis, where carcinogenesis is caused by a sequence of mutations that create a rapidly proliferating cell type that evades normal checks on cell growth. This creates a positive feedback cycle, where unchecked cell growth decreases genomic stability and permits further mutation (Lodish et al. 2003). Indeed, as tumors develop, cancer cells acquire mutations that differentiate them from their progenitors, leading to differences in treatment responses within cells of the same tumor (Croce 2008). Additionally, the“multi-hit” hypothesis explains why most cancers develop later in life, as a perfect storm of mutations steadily acquired through life are required to initiate cancer development (Lodish et al. 2003).
How do mutations to DNA give rise to a cancerous phenotype? Altering the DNA sequence of genes can lead to aberrant proteins which disregu- late important pathways involving cell proliferation and survival. There are six main phenotypic hallmarks of cancer: 1) loss of contact inhibition, 2) exaggerated response to growth regulating signals, 3) failure to undergo pro- grammed cell death in response to genetic damage, 4) immortalization 5) ability to evade immune defenses and 6) the production of factors promoting increased vascularization of the tumor (angiogenesis) (Hanahan & Weinberg 2000).
Genes involved in cancer induction can be classified into two major cat- egories, proto-oncogenes and tumor suppressor genes. Both proto-oncogenes and tumor suppressor gene classes are involved in the maintenance of the cell cycle, or the endogenous system that controls cell division. Mutations to proto-oncogenes are typically dominant, gain of function mutations; that is, mutation to only one genomic copy of the gene is sufficient to induce cancer (Lodish et al. 2003). When a proto-oncogene is mutated, it becomes “activated” to an oncogene which can contribute to carcinogenesis. Tumor suppressor genes are a broad class of genes that encode cell cycle inhibitors, receptor proteins for inhibitory hormones, checkpoint proteins that halt the cell cycle upon DNA damage, and proteins that promote apoptosis (Lodish et al. 2003). Under normal function, tumor suppressor genes act to restrain cell division. Mutation to these genes permits inappropriate growth. Muta- tions in tumor suppressors are typically recessive, loss of function mutations, meaning that mutations (hits) to both gene copies are required to inactivate tumor suppressor genes (Lodish et al. 2003). An important subclass of tumor suppressor genes are DNA repair genes, which are responsible for ensuring genomic stability. In short, DNA repair genes fix DNA damaged by copying errors or mutagens, preventing the accumulation of potentially deleterious or cancerous mutations.
A useful analogy for carcinogenesis is the progressive breakdown of a car. In a normal human cell (much like a functioning car), there are multiple layers of control to ensure the cell operates predictably. Proto-oncogenes, like the gas pedal of the cell, provide the go signals for cell division. Tumor suppressor genes, like the breaks, halt cell growth until appropriate conditions for division are met. DNA repair genes, like your mechanic, ensure that the cellular machinery are functioning properly. All of these components work in concert to tightly control the cell’s passage through the cell cycle. But over time, an accumulation of inherited and acquired defects (like manufacturing errors and wear and tear on a car) can cause the internal checks on cell growth to fail. This can causes the cell to divide uncontrollably resulting in tumor growth. Figure 1.1: Figure from Lawrence et al. (2013). Main panel: frequency of mutation (# mutations/megabase) illustrates remarkable heterogeneity in mutation rates between and within cancer types. Minor panel: relative proportion of base substitution signatures within and between cancer types. While mutation is the unifying molecular cause of cancer, it is not a monolithic process. The types and frequencies of mutation vary dramatically across the genome and across different cancer types. Some mutations are highly specific to particular cancer types, while others are present in many cancers. For example, the gene TP53 is found mutated at high frequencies in many cancers, while the BRCA1 gene is highly specific to breast and ovarian cancers (Schaefer & Serrano 2016).
In a study by Lawrence et al. (2013), the exomes (protein coding regions of the genome) of 3083 tumor samples spanning 27 different cancer types were sequenced. The authors assessed mutational heterogeneity on several differ- ent levels: mutation frequencies between tumors of the same cancer type, mutation frequencies between tumors of different cancer types, mutation sig- natures between tumors of different cancer types, and mutation frequencies between different locations of the genome. Analysis of the 27 cancer types revealed that median mutation frequencies varied up to 1000-fold between cancer types as seen in the main panel of Figure 1.1. About half the variation in mutation frequency was explainable by cancer type. The study also found marked variation in mutation frequency between tumors of the same cancer types, also visible on the main panel of Figure 1.1. For example, the cancers with the highest mutation frequencies are lung cancers and melanomas, due to the potency of the carcinogens at work (cigarette smoke and UV radiation respectively) (Vogelstein et al. 2013). The results of Lawrence et al. (2013) demonstrate the remarkable variability in mutation frequency between different types of cancer and between different patients.
More complex mutational phenomena also show tissue specific patterns. The frequency of catastrophic “hypermutation” events are known to occur with varying frequencies in different cancer types. Complex mutation rela- tionships such as co-mutation or mutual exclusivity are also known to be remarkably specific to certain cancer types (Schneider et al. 2018). Studies of somatic mutation have illustrated remarkable heterogeneity in mutational phenomena, which can be largely explained by differences in tissue type and background mutation rate. Figure 1.2: Mutation spectra in different cancer types reveals rich variation among different cancer types and natural grouping among cancers of the same types. Distance from the center represents the total mutation frequency (in mutations/Mb) while the angle indicates the relative contributions of each spectra (Lawrence et al. 2013).
Different cancer types also show marked variability in mutation spectra, or relative contributions of each single base substitution signature in the can- cer genome. The minor panel of Figure 1.1 shows the relative proportions of the 6 possible single nucleotide base substitution signatures. A remark- able example is the last three panels, which show the mutation signature contributions in Lung adenocarcinoma, Lung squamous cell carcinoma, and Melanoma. In the two lung cancers, the majority of mutations are C→A changes, as indicated by the large blue shaded area. These mutations are consistent with exposure to the hydrocarbons found in cigarette smoke. In contrast, the melanomas were composed of an overwhelming number of C→T mutations, consistent with exposure to excess UV radiation. The heterogeneity in mutation signatures between cancer types is further
illustrated in the radial plot in Figure 1.2. A dimension reduction algorithm
(non-negative matrix factorization) was used to summarize each tumor sam-
ple according to the relative contributions of each of the 96 single nucleotide mutations signatures.  Plotting tumors according to their mutation frequen- cies and the relative contributions of the mutation spectra identified by the algorithm illustrates a grouping of tumors according to their cancer type. Figure 1.2 illustrates a link between relative contributions of different muta- tion types and cancer type. Lastly, the study by Lawrence et al. (2013) identified that mutation fre- quency varied regionally across the genome, with differences in mutation frequency varying between 5-fold and 10-fold at different positions along the genome. The authors contended that a failure to account for mutational heterogeneity (and therefore background mutation rates) was a major hin- drance to identifying mutations that were statistically significant from the background rate (and therefore relevant to cancer). In short, mutation is not a monolithic process in human cancer. Mutation varies between cancer patients, cancer types, and within the genome itself. Harnessing rare mutations: a clinically- relevant endeavor. In addition to improving understanding of cancer etiology, biomolecular study of cancer genomes can help inform diagnosis and patient treatment. The idea underlying the field of precision oncology is that molecular characterization of a patient’s tumor can help predict an individual patient’s response to specific treatments. These genotype-directed therapies have led to dramatic improvements in patient outcomes, providing a positive sign for the future of personalized cancer medicine (Scholl & Fro 2019).
Comprehensive sequencing efforts to uncover the mutational drivers of
human cancer are complicated by a “long-tail” phenomenon, where sparsely mutated genes vastly outnumber commonly mutated ones (Vogelstein et al. 2013). Sequencing efforts have documented a small number of “mountains” and a large number of “hills”, i.e., a small number genes that acquire a large number of mutations, and a large number of genes that are mutated infrequently (Vogelstein et al. 2013). Mutational “mountains”, such as genes like TP53 and KRAS, have been studied intensively, are extremely likely to drive carcinogenesis, and in some cases have associated treatments designed to target the particular gene or gene product. However, the challenge lies in the many “hills”. Sparsely mutated genes dominate the landscape of muta- tion in cancer, and several likely play important roles in cancer development. A good example is the rare gene fusions in the NTRK family of receptor kinases, which allow cancers to evade normal checks on cell growth (Scholl & Fro 2019). This implies that many clinically-relevant yet rare mutations remain undiscovered. Thus, the statistical task of distinguishing sparsely mu- tated cancer genes from the background mutational noise is both extremely important and difficult.
Several different approaches have been used to identify and illustrate the clinical-relevance of rare mutations. One large scale analysis of nearly 25,000 cancers identified over 1000 mutational hotspots, of which 26% were novel (Chang et al. 2018). This brute force approach demonstrated the importance of large genome sequencing efforts sufficiently powered to identify rare driver variants. A clever statistical approach to identifying rare cancer drivers in- volves identifying mutational signals among proximal groups of “hills”. For example, the HotNet2 model (Leiserson et al. 2015) assembles genes into small interacting networks and identifies rare drivers by identifying subnet- works with substantially higher-than-expected mutation scores. HotNet2 groups individual “mountains” into mountain ranges and their associated “foothills”, enabling more powerful identification of rare driver genes. Cancers of unknown primary site, clonal origin of metastasis, and liquid biopsy. Study of mutational heterogeneity of different cancer types (especially rare mutations) is relevant to clinical problems like diagnosing cancers of unknown primary site, identifying the clonal origin of a metastatic cancer, and in emerging liquid biopsy technologies. Cancers of unknown primary. Cancers of unknown primary site (CUPs) comprise 3-5% of cancer diagnoses globally and are typically associated with poor health outcomes (Pavlidis & Khaled 2015). Cancers of unknown primary site are often metastatic cancers for which the anatomical site of origin is unknown after detailed investiga- tion. Once thought to be their own class of cancers, it is now recognized that CUPs are a heterogeneous class of tumors, each retaining a genomic signature of their tissue of origin. The advent of improved imaging tech- niques, immunohistochemical testing, and genomic and proteomic sequenc- ing tools have sophisticated our approach to diagnosing and treating these cancers. Thus, cancers of unknown primary are a prime target for person- alized medicine, as treatment can be informed by the molecular profile of individual patients (Varadhachary & Raber 2014). Clonal origin of metastasis.A major challenge for pathologists is determining whether a tumor is a metastatic tumor or independent occurrence. This task is typically done by comparing the histological characteristics of tumor cells, but genetic mark- ers have become relevant to this decision. The idea is that two tumors that metastatically evolved from the same progenitor will have some somatic mu- tations in common, while unrelated tumors will display different mutation profiles. However, the significance of shared mutation depends precisely on how common the observed mutations are. Thus, using genomic data, par- ticularly rare variant data, to test for clonal relatedness is a important task (Ostrovnaya et al. 2015). Liquid biopsy. Liquid biopsy is the analysis of cell-free, circulating tumor DNA obtained through a minimally invasive procedure such as a blood draw or urine sam- ple (Wan et al. 2017). Liquid biopsies could be clinically beneficial in a vari- ety of settings: early or population-wide cancer detection screens, prognosis assessment, treatment selection, and treatment monitoring. Liquid biopsy approaches are still an emerging technology, and identifying better ways to match mutational information in circulating tumor DNA fragments to spe- cific cancer diagnoses will render these tests more efficacious and more widely used. Summary. The biomedical goal of this thesis is to improve extraction of cancer type specific signals from large-scale sequencing data (particularly from sparsely mutated genes) in the cancer genome. These signals could potentially be relevant to the clinical tasks outlined in the previous section. The major statistical challenge, of which the remainder of my thesis will attempt to address, involves harnessing the signal encoded in the preponderance of rare mutation in the cancer genome.
"

Clean_String <- function(string){
  # Lowercase
  temp <- tolower(string)
  # Remove everything that is not a number or letter (may want to keep more 
  # stuff in your actual analyses). 
  temp <- stringr::str_replace_all(temp,"[^a-zA-Z\\s]", " ")
  # Shrink down to just one white space
  temp <- stringr::str_replace_all(temp,"[\\s]+", " ")
  # Split it
  temp <- stringr::str_split(temp, " ")[[1]]
  # Get rid of trailing "" if necessary
  indexes <- which(temp == "")
  if(length(indexes) > 0){
    temp <- temp[-indexes]
  } 
  return(temp)
}

txt<-Clean_String(txt)

res<-txt %>% table() %>% table()
res<-data.frame(r=as.numeric(names(res)), N_r=as.numeric(res))
res
png("textex.png", height=4.75, width=4, units="in", res=300)
res %>% ggplot(aes(x=r, y=N_r))+geom_histogram(stat="identity", fill="#619CFF")+scale_y_continuous(trans=scales::pseudo_log_trans(base=10), breaks=c(0, 10, 100, 1000))+theme_minimal()+theme(axis.text=element_text(size=14), axis.title=element_text(size=16), panel.border = element_rect(colour = "grey", fill=NA, size=1))+ggtitle("Words in my first thesis chapter (r, N_r)")
dev.off()
